go-for-eat/navigation/index.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { addNavigationHelpers, StackNavigator } from 'react-navigation';
+
+import { NavBar } from '../components/NavBar';
+import Home from '../screens/Home/Home';
+import Login from '../screens/Login/Login';
+import User from '../screens/User';
+import CreateEvent from '../screens/CreateEvent';
+import LoadingPage from '../screens/LoadingPage';
+
+const NavigatorWithRootScreen = (name) => {
+  return StackNavigator({
+    [name] : {
+      screen:name,
+      navigationOptions:{
+        gesturesEnabled:false,
+        header:(<NavBar/>),
+      }
+    }
+  });
+};
+
+const NavigatorWithModal = (name) => {
+  return StackNavigator({
+    [name] : {
+      screen:name,
+      navigationOptions:{
+        gesturesEnabled:false,
+      }
+    },
+  },
+  {
+    mode: 'modal',
+    headerMode: 'none'
+  });
+};
+
+const HomeStack = StackNavigator({
+  Home: {
+    screen: NavigatorWithRootScreen(Home),
+    navigationOptions:{header:()=>null}
+  },
+
+  User: {
+    screen: NavigatorWithRootScreen(User),
+    navigationOptions:{header:()=>null}
+  },
+  CreateEvent: {
```

Comment:
Sorry, doesn't it mean that CreateEvent is now part of the `HomeStack` so it will be pushed when presented?

Reply:
I imagined it to be something like
`NavigatorWithModal(NavigatorWithRootScreen(MainNavigator), CreateEvent)`